src/enable_variable_ops/ngraph_remove_ngraphassigns.cc

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_remove_ngraphassigns.h"
+
+using namespace std;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+Status RemoveNGraphAssigns(Graph* graph) {
+  vector<Node*> remove_nodes;
+
+  for (auto node : graph->op_nodes()) {
+    if (node->type_string() == "NGraphAssign" && NodeIsMarkedForRemoval(node)) {
+      NGRAPH_VLOG(3) << "Removing NGraphAssign " << node->name();
+      Node *input_0, *input_1;
+      // input_0 is of type NGraphVariable/NGraphAssign
+      // input_1 is of type NGraphEncapsulateOp
+      TF_RETURN_IF_ERROR(node->input_node(0, &input_0));
+      TF_RETURN_IF_ERROR(node->input_node(1, &input_1));
```

Comment:
Is there a way to validate this assumption?

Reply:
The "_ngraph_remove" is attached to only NGraphAssign nodes whose input 1 is from NGraphEncapsulate Ops. 
I can add a check here. The tests will need changes.